chipsec/helper/linuxnative/linuxnativehelper.py

```diff
@@ -0,0 +1,462 @@
+                                "Are you running this command as root?\n"
+                                f"{str(err)}", err.errno)
+
+    def devmsr_available(self) -> bool:
+        """Check if /dev/cpu/CPUNUM/msr is usable.
+           In case the driver is not loaded, we might be able to perform the
+           requested operation via /dev/cpu/CPUNUM/msr. This requires loading
+           the (more standard) msr driver. Returns True if /dev/cpu/CPUNUM/msr
+           is accessible.
+        """
+        if self.dev_msr:
+            return True
+
+        try:
+            self.dev_msr = {}
+            if not os.path.exists("/dev/cpu/0/msr"):
+                os.system("modprobe msr")
+            for cpu in os.listdir("/dev/cpu"):
+                logger().log_debug(f"found cpu = {str(cpu)}")
+                if cpu.isdigit():
+                    cpu = int(cpu)
+                    self.dev_msr[cpu] = os.open(f"/dev/cpu/{str(cpu)}/msr", os.O_RDWR)
+                    logger().log_debug(f"Added dev_msr {str(cpu)}")
+            return True
+        except IOError as err:
+            raise OsHelperError("Unable to open /dev/cpu/CPUNUM/msr.\n"
+                                "This command requires access to /dev/cpu/CPUNUM/msr.\n"
+                                "Are you running this command as root?\n"
+                                "Do you have the msr kernel module installed?\n"
+                                f"{str(err)}", err.errno)
+
+    def close(self):
+        if self.dev_mem:
+            os.close(self.dev_mem)
+        self.dev_mem = None
+        
+    def getcwd(self) -> str:
+        return(os.getcwd())
+
+###############################################################################################
+# Actual API functions to access HW resources
+###############################################################################################
+
+    def read_pci_reg(self, bus: int, device: int, function: int, offset: int, size: int, domain: int = 0) -> int:
+        device_name = "{domain:04x}:{bus:02x}:{device:02x}.{function}".format(
+                      domain=domain, bus=bus, device=device, function=function)
+        device_path = f"/sys/bus/pci/devices/{device_name}/config"
+        if not os.path.exists(device_path):
+            if offset < 256:
+                if self.legacy_pci:
+                    pci = self.legacy_pci
+                else:
+                    pci = LEGACY_PCI()
+                    self.legacy_pci = pci
+                value = pci.read_pci_config(bus, device, function, offset)
+                if size == 1:
+                    value = value & 0xFF
+                elif size == 2:
+                    value = value & 0xFFFF
+                elif size == 4:
+                    value = value & 0xFFFFFFFF
+
+                return value
+            else:
+                byte = b"\xff"
+                return defines.unpack1(byte * size, size)
```

Comment:
Will this lead to false results?

Reply:
Would it be better to raise an exception in this case?